File: tests/fixtures/codegen/dimensions/pointers-expected/typescript/runtime/queries/arrays/SharedDimensions.ts
import type { SharedDimensionsParams } from "../../../types/queries/arrays/SharedDimensions.js";
export type { SharedDimensionsParams } from "../../../types/queries/arrays/SharedDimensions.js";
import type * as pgsid from "./pgsid/queryable.js";
export const sharedDimensionsSql = "INSERT INTO arrays (id, ordinary, matrix, flexible, owners)\nVALUES (1, ARRAY[1], $1, $1, ARRAY[1]::user_id[]);";
export async function sharedDimensions(db: pgsid.Queryable, params: SharedDimensionsParams): Promise<void> {
    await db.query(sharedDimensionsSql, [params["shared"]]);
}

File: tests/fixtures/codegen/dimensions/pointers-expected/queries/arrays/shareddimensions.go
package arrays

import (
	"context"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
	"github.com/jackc/pgx/v5/pgtype"
)

const SharedDimensionsSQL = "INSERT INTO arrays (id, ordinary, matrix, flexible, owners)\nVALUES (1, ARRAY[1], $1, $1, ARRAY[1]::user_id[]);"

type SharedDimensionsParams struct {
	Shared pgtype.Array[int32] `db:"shared"`
}

func (q *Queries) SharedDimensions(ctx context.Context, params SharedDimensionsParams) error {
	arrayInput1, err := pgsidpgx.ValidateArrayInput[int32](params.Shared, []int{2}, "SharedDimensions", "shared", false)
	if err != nil {
		return err
	}
	_, err = q.db.Exec(ctx, SharedDimensionsSQL, arrayInput1)
	return err
}
